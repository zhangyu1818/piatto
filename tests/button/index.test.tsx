import React from 'react'
import { mount, render } from 'enzyme'
import Button from '../../components/button'
import mountTest from '../shared/mountTest'

describe('Button', () => {
  mountTest(Button)

  it('renders correctly', () => {
    expect(<Button>piatto</Button>).toMatchSnapshot()
  })

  it('should render empty button without errors', () => {
    const wrapper = render(
      <Button>
        {null}
        {undefined}
      </Button>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should be clickable', () => {
    const onClickMock = jest.fn()
    const wrapper = mount(<Button onClick={onClickMock} />)
    wrapper.simulate('click')
    expect(onClickMock).toHaveBeenCalled()
  })

  it('should support to change loading', () => {
    const wrapper = mount(<Button />)
    expect(wrapper.find('.piatto-button-loading').length).toBe(0)
    wrapper.setProps({ loading: true })
    expect(wrapper.find('.piatto-button-loading').length).toBe(1)
    wrapper.setProps({ loading: false })
    expect(wrapper.find('.piatto-button-loading').length).toBe(0)
  })

  it('should not clickable when button is loading', () => {
    const onClickMock = jest.fn()
    const wrapper = mount(<Button loading onClick={onClickMock} />)
    wrapper.simulate('click')
    expect(onClickMock).not.toHaveBeenCalled()
  })
})
