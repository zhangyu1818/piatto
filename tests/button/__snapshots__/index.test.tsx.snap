// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Button>
  piatto
</Button>
`;

exports[`Button should render empty button without errors 1`] = `ReactWrapper {}`;
